src/Infrastructure/Test/Repositories/InMemoryTaxonomyRepository.php

```
@@ -0,0 +1,84 @@
+    }
+
+    private function existsByKey(TaxonomyKeyId $taxonKeyId, TaxonomyId $allowedTaxonomyId): bool
+    {
+        foreach (self::$taxonomies as $taxonomy) {
+            if (! $taxonomy->taxonomyId->equals($allowedTaxonomyId) && $taxonomy->hasTaxonomyKeyId($taxonKeyId)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'existsByKey'. (Unused Code Rules)

(UnusedPrivateMethod)